### PR TITLE
Testing Adding CI

### PR DESCRIPTION
**Pull request by CSSFrancis**
Testing adding in CI similar to kikchipy's testing of their documentation

### CODE REVIEW

**Comment by CSSFrancis** [comment]
This doesn't necessarily fix 02 and 11 but it does help to test to make sure that the demos are running and up to date.

Maybe testing these in the main pyxem repo makes more sense than testing that the notebooks run here but something is better than nothing.

**Comment by CSSFrancis** [comment]
Reference to #82

**Comment by CSSFrancis** [comment]
Okay the only notebook that isn't passing now is 02.  That one might take a bit more effort and potentially a bug fix upstream to pyxem.  It appears that a change in `diffsims` wasn't properly being tested in `pyxem` so I'll need to make that bug fix.

File: run_nbval.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# This script must be run from the pyxem-demos/ top directory:
+#   $ chmod u+x ./doc/tutorials/run_nbval.sh
+#   $ ./doc/tutorials/run_nbval.sh
```
Comment (hakonanes) [inline run_nbval.sh:5]:
```suggestion
#   $ chmod u+x run_nbval.sh
#   $ ./run_nbval.sh
```

File: run_nbval.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  "12 MolecularGlassOrientationMapping.ipynb"\
+)
+
+# Append relative path to notebook names
+for i in "${!NOTEBOOKS[@]}"; do
+  NOTEBOOKS[i]="${NOTEBOOKS[i]}"
+done
+
+# Test with nbval
+pytest -v --tb=auto --nbval "${NOTEBOOKS[@]}" --nbval-lax --current-env
```
Comment (hakonanes) [inline run_nbval.sh:29]: See [CI log](https://github.com/pyxem/pyxem-demos/pull/85/checks#step:6:13)

```suggestion
pytest -v --tb=auto --nbval "${NOTEBOOKS[@]}" --nbval-lax --nbval-current-env
```

Is not `auto` the default traceback style?

File: run_nbval.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  "09 Angular Correlations of Amorphous Materials.ipynb"\
+  "10 STEM DPC Analysis of Magnetic Sample.ipynb"\
+  "11 Accelerated orientation mapping with template matching.ipynb"\
+  "12 MolecularGlassOrientationMapping.ipynb"\
+)
+
+# Append relative path to notebook names
+for i in "${!NOTEBOOKS[@]}"; do
+  NOTEBOOKS[i]="${NOTEBOOKS[i]}"
+done
```
Comment (hakonanes) [inline run_nbval.sh:26]: This isn't necessary here:

```suggestion
```

**Review by hakonanes — APPROVED** [review]
I find testing of docs *very* important. Happy to see this PR. And that you could re-use some stuff from kikuchipy.

Made a few comments.

**Pull request merged**